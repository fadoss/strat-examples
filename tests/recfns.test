semantics/recfns

srew 'f(10) using redValue(factorial) .
srew 'f(10) using redName(factorial) .
srew 'A(1, 3) using redName(ackermann) .
srew 's(2) using redValue(sign) .
srew 's(-2) using redValue(sign) .
srew 'g('f(3)) using redName(nameonly) .

