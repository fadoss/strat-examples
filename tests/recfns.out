==========================================
srewrite in MAIN : 'f(10) using redValue(factorial) .

Solution 1
rewrites: 308
result NzNat: 3628800

No more solutions.
rewrites: 308
==========================================
srewrite in MAIN : 'f(10) using redName(factorial) .

Solution 1
rewrites: 308
result NzNat: 3628800

No more solutions.
rewrites: 308
==========================================
srewrite in MAIN : 'A(1, 3) using redName(ackermann) .

Solution 1
rewrites: 730
result NzNat: 5

No more solutions.
rewrites: 730
==========================================
srewrite in MAIN : 's(2) using redValue(sign) .

Solution 1
rewrites: 199
result NzNat: 1

No more solutions.
rewrites: 199
==========================================
srewrite in MAIN : 's(-2) using redValue(sign) .

Solution 1
rewrites: 202
result NzInt: -1

No more solutions.
rewrites: 202
==========================================
srewrite in MAIN : 'g('f(3)) using redName(nameonly) .

Solution 1
rewrites: 8
result NzNat: 7

No more solutions.
rewrites: 8
Bye.
